great_expectations/validator/validation_graph.py

```diff
@@ -28,20 +40,190 @@ def id(self):
+
+            # Check to see if the user has disabled progress bars
+            disable = not show_progress_bars
+            if len(self.edges) < min_graph_edges_pbar_enable:
+                disable = True
+
+            if progress_bar is None:
+                # noinspection PyProtectedMember,SpellCheckingInspection
+                progress_bar = tqdm(
+                    total=len(ready_metrics) + len(needed_metrics),
+                    desc="Calculating Metrics",
+                    disable=disable,
+                )
+            progress_bar.update(0)
+            progress_bar.refresh()
+
+            computable_metrics = set()
+
+            for metric in ready_metrics:
+                if metric.id in failed_metric_info and failed_metric_info[metric.id]["num_failures"] >= MAX_METRIC_COMPUTATION_RETRIES:  # type: ignore[operator]  # Incorrect flagging of 'Unsupported operand types for <= ("int" and "MetricConfiguration") and for >= ("Set[ExceptionInfo]" and "int")' in deep "Union" structure.
+                    aborted_metrics_info[metric.id] = failed_metric_info[metric.id]
+                else:
+                    computable_metrics.add(metric)
+
+            try:
+                # Access "ExecutionEngine.resolve_metrics()" method, to resolve missing "MetricConfiguration" objects.
+                metrics.update(
+                    self._execution_engine.resolve_metrics(
+                        metrics_to_resolve=computable_metrics,
+                        metrics=metrics,
+                        runtime_configuration=runtime_configuration,
+                    )
+                )
+                progress_bar.update(len(computable_metrics))
+                progress_bar.refresh()
+            except ge_exceptions.MetricResolutionError as err:
+                if catch_exceptions:
+                    exception_traceback = traceback.format_exc()
+                    exception_message = str(err)
+                    exception_info = ExceptionInfo(
+                        exception_traceback=exception_traceback,
+                        exception_message=exception_message,
+                    )
+                    for failed_metric in err.failed_metrics:
+                        if failed_metric.id in failed_metric_info:
+                            failed_metric_info[failed_metric.id]["num_failures"] += 1  # type: ignore[operator]  # Incorrect flagging of 'Unsupported operand types for <= ("int" and "MetricConfiguration") and for >= ("Set[ExceptionInfo]" and "int")' in deep "Union" structure.
+                            failed_metric_info[failed_metric.id]["exception_info"].add(exception_info)  # type: ignore[union-attr]  # Incorrect flagging of 'Item "MetricConfiguration" of "Union[MetricConfiguration, Set[ExceptionInfo], int]" has no attribute "add" and Item "int" of "Union[MetricConfiguration, Set[ExceptionInfo], int]" has no attribute "add"' in deep "Union" structure.
```

Comment:
I wouldn't say the type system is incorrect here, it's just that we haven't proven to the typing system that this item can't be an `int` etc.

Reply:
@Kilo59 Thanks -- as agreed, we can address in a follow-on PR.